### PR TITLE
Update docs to reflect default behavior if PARTITIONS property not supplied.

### PR DESCRIPTION
**Pull request by big-andy-coates**
Fix for #672. 

The default is that the partition count is controlled by `KsqlConfig.SINK_NUMBER_OF_PARTITIONS_PROPERTY`

### CODE REVIEW

File: docs/syntax-reference.rst [hunk 1]
```diff
@@ -247,17 +247,21 @@ The WITH clause supports the following properties:
...
-| PARTITIONS   | The number of partitions in the topic. If this        |
-|              | property is not set, then the number of partitions of |
-|              | the input stream/table will be used.                  |
+| PARTITIONS   | The number of partitions in the topic.                |
+|              | If this property is not set, then the number of       |
+|              | partitions is taken from the value of the             |
+|              | ``ksql.sink.partitions`` property.                    |
+|              | ``ksql.sink.partitions`` can be set in the properties |
+|              | file the KSQL server is started with, or by using the |
+|              | ``SET`` statement.                                    |
```
Comment (hjafarpour) [inline docs/syntax-reference.rst]: You may want to also mention that the default value for `ksql.sink.partitions` is 4.

File: docs/syntax-reference.rst [hunk 2]
```diff
@@ -303,17 +307,21 @@ The WITH clause supports the following properties:
...
```
Comment (hjafarpour) [inline docs/syntax-reference.rst]: As above the default is 4.

File: docs/syntax-reference.rst (hunk 1 above)
Comment (big-andy-coates) [reply docs/syntax-reference.rst]: Good call. Consider it done.

File: docs/syntax-reference.rst (hunk 2 above)
Comment (big-andy-coates) [reply docs/syntax-reference.rst]: as above.

**Pull request merged**